### PR TITLE
Fix CockroachDB errors

### LINKED ISSUE #979: Cockroach DB errors
#942 added the following SQL statement:

```crystal
db.exec "CREATE DATABASE #{db_name} WITH OWNER DEFAULT"
```

This results in the following error in Cockroach DB:

```
at or near "default": syntax error (PQ::PQError)
  from lib/pg/src/pq/connection.cr:214:7 in 'handle_error'
  from lib/pg/src/pq/connection.cr:197:7 in 'handle_async_frames'
  from lib/pg/src/pq/connection.cr:173:7 in 'read'
  from lib/pg/src/pq/connection.cr:168:7 in 'read'
  from lib/pg/src/pq/connection.cr:446:31 in 'expect_frame'
  from lib/pg/src/pq/connection.cr:445:5 in 'expect_frame'
  from lib/pg/src/pg/statement.cr:18:5 in 'perform_query'
  from lib/pg/src/pg/statement.cr:35:14 in 'perform_exec'
  from lib/db/src/db/statement.cr:85:9 in 'perform_exec_and_release'
  from lib/db/src/db/statement.cr:70:7 in 'exec:args'
  from lib/db/src/db/query_methods.cr:275:7 in 'exec'
  from lib/avram/src/avram/migrator/runner.cr:60:7 in 'create_db'
  from lib/avram/src/avram/tasks/db/create.cr:20:5 in 'run_task'
  from lib/avram/src/avram/tasks/db/base_task.cr:6:7 in 'call'
  from spec/setup/database.cr:2:3 in '__crystal_main'
  from /home/runner/work/_temp/crystal-1.6.0-true-undefined/share/crystal/src/crystal/main.cr:115:5 in 'main_user_code'
```

Removing `WITH OWNER DEFAULT` works OK:

```crystal
db.exec "CREATE DATABASE #{db_name}"
```

There's also an error with `Avram::Database::DatabaseCleaner#truncate`:

```crystal
statement = ("TRUNCATE TABLE #{table_names.map { |name| name }.join(", ")} RESTART IDENTITY CASCADE;")
```

Removing `RESTART IDENTITY` works OK:

```crystal
statement = "TRUNCATE TABLE #{table_names.join(", ")} CASCADE;"
```

### PR DESCRIPTION
**Pull request by akadusei**
Fixes #979

### CODE REVIEW

**Review by jwoertink — APPROVED** [review]

File: src/avram/database.cr [hunk 1]
```diff
@@ -263,7 +263,7 @@ abstract class Avram::Database
     def truncate
       return if table_names.empty?
 
-      statement = ("TRUNCATE TABLE #{table_names.map { |name| name }.join(", ")} RESTART IDENTITY CASCADE;")
```
Comment (rmarronnier) [inline src/avram/database.cr:266]: From https://www.postgresql.org/docs/current/sql-truncate.html :
> RESTART IDENTITY
Automatically restart sequences owned by columns of the truncated table(s).

Could we keep the original behavior maybe as a method argument ?
`def truncate(restart_identity = true)`

Comment (jwoertink) [reply src/avram/database.cr:266]: oh, that's a good point :thinking: This all happens in a few different areas

Truncating the database
https://github.com/luckyframework/avram/blob/df3c20f195aa6f844372fa1089ec71fe55f47399/src/avram/database.cr#L43-L46

Truncating a table
https://github.com/luckyframework/avram/blob/df3c20f195aa6f844372fa1089ec71fe55f47399/src/avram/queryable.cr#L54-L61

Maybe we need to add for both?

Comment (akadusei) [reply src/avram/database.cr:266]: That's probably a better idea. It maintains the original behaviour by default, just in case someone depended on it.

Comment (akadusei) [reply src/avram/database.cr:266]: I could push up subsequent commits with the requested changes, if you're up for it?

> Truncating a table

Should this be a separate PR, or bundling it with this is OK?

Comment (jwoertink) [reply src/avram/database.cr:266]: I'm ok with it being bundled. Once this is all merged in, I'll do a quick Avram release.

**Review by jwoertink — APPROVED** [review]

**Pull request merged**